mods/default/torch.lua

```diff
@@ -35,6 +35,26 @@ See LICENSE.txt and http://www.gnu.org/licenses/lgpl-2.1.txt
 
 --]]
 
+local function on_flood(pos, oldnode, newnode)
+	-- Instantiate item drop.
+	minetest.add_item(pos, ItemStack("default:torch 1"))
+
+	-- Play sound if torch is not flooded by igniter type liquid.
+	local nodedef = minetest.registered_items[newnode.name]
+	if nodedef == nil or not (
+			nodedef.groups ~= nil and
+			nodedef.groups.igniter ~= nil and
+			nodedef.groups.igniter > 0) then
```

Comment:
```lua
if not (nodedef and nodedef.groups and nodedef.groups.igniter and nodedef.groups.igniter) then
```

equivalent under de Morgan’s Theorem and Double Negation.
`~= nil` is redundant as it's not a boolean value.
`a == nil` can be replaced with `not a`.